language/stdlib/modules/libra_system.mvir

```diff
@@ -17,6 +17,12 @@ module LibraSystem {
         new_validator_set: Vector.T<Self.ValidatorInfo>,
     }
 
+    struct NewBlockEvent {
+        height: u64,
+        // On-chain time during the block at the given height
+        time_microseconds: u64,
```

Comment:
can we add the proposer and votes too? 😄

Reply:
> any blockers on this one? I'm looking at using block metadata for leader reputation too

Lemme take over the work! The main blocker was that we were migrating the genesis from using Move IR to source lang. Will put out a PR for this shortly.